### PR TITLE
[WIP] Retry Bucket#objects() when invalid XML returned

### PR DESCRIPTION
**Pull request by shimpeko**
Bucket#objects may throw `Aws::Xml::Parser::ParsingError` when S3 API returned invalid XML.
ref. http://docs.aws.amazon.com/AmazonS3/latest/API/RESTBucketGET.html

 I added retry block since this exception can be retried.

### CODE REVIEW

File: lib/bricolage/s3datasource.rb [hunk 1]
```diff
@@ -125,7 +125,15 @@ def path(rel, no_prefix: false)
     end
 
     def traverse(rel, no_prefix: false)
+      retries ||= client.config.retry_limit
```
Comment (aamine) [inline lib/bricolage/s3datasource.rb]: 普通にこの行をrescueの外にすべき

File: lib/bricolage/s3datasource.rb [hunk 2]
```diff
@@ -125,7 +125,15 @@ def path(rel, no_prefix: false)
...
       bucket.objects(prefix: path(rel, no_prefix: no_prefix))
+    rescue Aws::Xml::Parser::ParsingError => e
+      if (retries -= 1) >= 0
```
Comment (aamine) [inline lib/bricolage/s3datasource.rb]: 行分ける

File: lib/bricolage/s3datasource.rb [hunk 3]
```diff
@@ -125,7 +125,15 @@ def path(rel, no_prefix: false)
...
+        retry
+        logger.warn "Retry Bucket#objects() for XML parsing error: #{e.message}"
+      else
+        throw
```
Comment (aamine) [inline lib/bricolage/s3datasource.rb]: `raise`

**Comment by shimpeko** [comment]
対応しました

**Pull request merged**